src/core/a-entity.js

```diff
@@ -427,18 +482,22 @@ var proto = Object.create(ANode.prototype, {
    */
   setEntityAttribute: {
     value: function (attr, oldVal, newVal) {
-      var component = components[attr];
-      oldVal = oldVal || this.getAttribute(attr);
-      // When creating entities programatically and setting attributes, it is not part
-      // of the scene until it is inserted into the DOM. This does not apply to scenes as
-      // scenes depend on its child entities to load.
-      if (!this.hasLoaded && !this.isScene) { return; }
+      if (components[attr]) {
+        this.updateComponent(attr, newVal);
+        return;
+      }
       if (attr === 'mixin') {
-        this.updateStateMixins(newVal, oldVal);
-        this.updateComponents();
+        this.mixinUpdate(newVal, oldVal);
         return;
       }
-      if (component) { this.updateComponent(attr, newVal); }
+    }
+  },
+
+  mixinUpdate: {
```

Comment:
updateFromMixins is not a good name either. This is the method triggered when there's a mixin attribute update (that triggers a components update)